widgets/src/widgets/cart.ts

```diff
@@ -0,0 +1,126 @@
+import { Widget } from '@/widgets/base'
+import {
+  WIDGET_COLORS,
+  AVAILABLE_STORE_CURRENCIES,
+  DEFAULT_CONTAINER_CSS_SELECTOR,
+} from '@/constants'
+
+import type { WidgetConfig } from '@/widgets/base'
+import type { OrderProduct, CartWidgetParams, CartWidgetRequestBody } from '@/interfaces'
+
+export class CartWidget extends Widget implements CartWidgetParams {
+  color
+  order
+  withPopup
```

Comment:
types are defined in the `implements CartWidgetParams` so any IDE can easily determine it, and this way we make sure we don't miss params on the constructor. This might not be perfect, but if I can, I like to lock the constructor to the class attritutes.
What do you mean by accessibility keys?

Reply:
yes, the types are defined by the interface, however even the IDE can't determine it, and i think it's just generally a good patter to implement the type also on the class, even if it's coming from the implemented interface
<img width="586" alt="Screenshot 2024-08-26 at 14 16 13" src="https://github.com/user-attachments/assets/faed8f11-8e1b-4513-a6bc-4fef1aa0ef68">